plugins/modules/system/sudoers.py

```diff
@@ -0,0 +1,190 @@
+            - This will be used for the filename for the sudoers file managed by this rule.
+        type: str
+    nopassword:
+        description:
+            - Whether a password will be required to run the sudo'd command.
+        default: true
+        type: bool
+    sudoers_path:
+        description:
+            - The path which sudoers config files will be managed in.
+        default: /etc/sudoers.d
+        type: str
+    state:
+        default: "present"
+        choices: [ present, absent ]
+        description:
+            - Whether the rule should exist or not.
+        type: str
+    user:
+        description:
+            - The name of the user for the sudoers rule.
+            - This option cannot be used in conjunction with I(group).
+        type: str
+'''
+
+EXAMPLES = '''
+- name: Allow the backup user to sudo /usr/local/bin/backup
+  sudoers:
+    name: allow-backup
+    state: present
+    user: backup
+    commands: /usr/local/bin/backup
+
+- name: Allow the monitoring group to run sudo /usr/local/bin/gather-app-metrics without requiring a password
+  sudoers:
+    name: monitor-app
+    group: monitoring
+    commands: /usr/local/bin/gather-app-metrics
+
+- name: Allow the alice user to run sudo /bin/systemctl restart my-service or sudo /bin/systemctl reload my-service, but a password is required
+  sudoers:
+    name: alice-service
+    user: alice
+    commands:
+      - /bin/systemctl restart my-service
+      - /bin/systemctl reload my-service
+    nopassword: false
+
+- name: Revoke the previous sudo grants given to the alice user
+  sudoers:
+      name: alice-service
+      state: absent
+'''
+
+import os
+from ansible.module_utils.basic import AnsibleModule
+
+
+class Sudoers(object):
+
+    def __init__(self, module):
+        self.check_mode = module.check_mode
+        self.name = module.params['name']
+        self.user = module.params['user']
+        self.group = module.params['group']
+        self.state = module.params['state']
+        self.nopassword = module.params['nopassword']
+        self.sudoers_path = module.params['sudoers_path']
+        self.file = os.path.join(self.sudoers_path, self.name)
+        self.commands = module.params['commands']
+
+    def write(self):
+        with open(self.file, 'w') as f:
+            f.write(self.content())
+
+    def delete(self):
+        os.remove(self.file)
+
+    def exists(self):
+        return os.path.exists(self.file)
+
+    def matches(self):
+        with open(self.file, 'r') as f:
+            return f.read() == self.content()
+
+    def content(self):
+        if self.user:
+            owner = self.user
+        elif self.group:
+            owner = '%{group}'.format(group=self.group)
+
+        commands_str = ', '.join(self.commands)
+        nopasswd_str = 'NOPASSWD:' if self.nopassword else ''
+        return "{owner} ALL={nopasswd} {commands}\n".format(owner=owner, nopasswd=nopasswd_str, commands=commands_str)
+
+    def run(self):
```

Comment:
Now that `check_mode` is an attribute of `Sudoers` you can go one step further and move the logic into `write` and `delete` directly. Specifically you can add guards at the top of each function like:
```python
def write(self):
    if self.check_mode:
        return
    ...
```

Reply:
Yeah, I like that.